app/components/custom-form/link-to-form-with-modal.js

```diff
@@ -0,0 +1,293 @@
+
+    // Insert
+    this.selectedInstanceOptions
+      .filter(
+        (selected) => !matches.find((m) => selected.instance.uri == m.value)
+      )
+      .forEach((opt) =>
+        updateSimpleFormValue(
+          this.storeOptions,
+          new NamedNode(opt.instance.uri)
+        )
+      );
+    super.updateValidations();
+  }
+
+  filterResults = task({ restartable: true }, async (event) => {
+    await timeout(INPUT_DEBOUNCE);
+    this.searchValue = event.target?.value;
+    this.pageToLoad = 0;
+    this.getInstances.perform();
+  });
+
+  get searchPlaceholderText() {
+    const labels = this.instanceDisplayLabels.map((d) => d.name);
+    return `Zoek op één van de eigenschappen van het formulier. Bv. ${labels.join(', ')}`;
+  }
+
+  get hasSelectedOptions() {
+    return this.selectedInstanceOptions?.length >= 1;
+  }
+
+  get formTypeUri() {
+    return this.storeOptions.store.any(
+      this.args.field.uri,
+      EXT('linkedFormType'),
+      undefined,
+      this.storeOptions.formGraph
+    )?.value;
+  }
+
+  get isFetchingInstances() {
+    return (
+      this.isSettingInitialSelectedOptions.isRunning ||
+      this.getInstances.isRunning
+    );
+  }
+
+  get instancesAsOptions() {
+    const instances = this.instancesToOptions(this.instances);
+
+    instances.meta = this.instancesMetadata;
+    return instances;
+  }
+
+  instancesToOptions(_instances) {
+    const keysOfLabels = this.instanceDisplayLabels.map((label) => label.name);
+    const uniqueInstanceIds = [];
+    const instances = [];
+    for (const instance of _instances) {
+      if (uniqueInstanceIds.includes(instance.id)) {
+        continue;
+      } else {
+        uniqueInstanceIds.push(instance.id);
+      }
+      instances.push({
+        option: keysOfLabels.map((key) => {
+          return {
+            key: key,
+            value: instance[key],
+          };
+        }),
+        instance,
+        isSelected: !!this.selectedInstanceOptions.find(
+          (o) => o.instance.id === instance.id
+        ),
+      });
+    }
+
+    return instances;
+  }
+
+  get filterInstancesParams() {
+    return {
+      page: this.pageToLoad,
+      size: 10,
+      filter:
+        this.searchValue && this.searchValue.trim() !== ''
+          ? this.searchValue
+          : null,
+    };
+  }
+
+  @action
+  fetchPreviousPage({ previousPage }) {
+    this.pageToLoad = previousPage;
+    this.getInstances.perform();
+  }
+
+  @action
+  fetchNextPage({ nextPage }) {
+    this.pageToLoad = nextPage;
+    this.getInstances.perform();
+  }
+
+  getInstances = task({ enqueue: true }, async () => {
+    if (!this.formTypeUri || isNaN(this.pageToLoad)) {
+      return;
+    }
+    const formInfo = await this.fetchInstancesForUris([]);
+    this.instanceDisplayLabels = formInfo.labels;
+    this.instances.clear();
+    this.instances.pushObjects(formInfo.instances);
+  });
+
+  isSettingInitialSelectedOptions = task(async () => {
```

Comment:
this reads like a getter, i would name the task something active like setInitialSelectedOptions